core/mutex.c

```diff
@@ -33,21 +33,20 @@
 #define ENABLE_DEBUG 0
 #include "debug.h"
 
-int mutex_lock(mutex_t *mutex)
+/**
+ * @brief   Block waiting for a locked mutex
+ * @pre     IRQs are disabled
```

Comment:
if this is a pre, why is the irq_state necessary? just hard enable IRQs afterwards.

Reply:
It's not :-/ Sneaked in.

Reply:
Wait, this is right. I just saw the `mutex_lock` and misread as if I added this to `mutex_lock()` somehow.

And `irq_state` is need in case someone does stuff like:

```C
    unsigned state = irq_disable();
    mutex_lock(&mut);
    other_stuff_with_irqs_disabled_i_know_this_is_stupid();
    irq_restore(state);    
```

I don't this that there is a valid use case for the above code. But IMO we shouldn't judge what application developers implement. And if the want to do stupid stuff, the should certainly be able to do stupid stuff in a safe manner.

Reply:
> I don't think that there is a valid use case for the above code.

no, it will probably horribly fail, as the mutex code expects to schedule away on irq_restore(). But that is not new to this PR: :/

Reply:
Fixing that would finally be a use case for the return value of `irq_enable()` :-D